couchbase_cluster_admin/client.py

```diff
@@ -1,18 +1,34 @@
+import time
+
 import requests
 
 
 class BaseClient:
-
     def http_request(self, url, method="GET", data=None, headers={}, timeout=10.0):
         auth = None
         if self.username is not None and self.password is not None:
             auth = (self.username, self.password)
 
-        return requests.request(
-            method,
-            url,
-            data=data,
-            headers=headers,
-            auth=auth,
-            timeout=timeout,
-        )
+        # Retrying POST requests could lead to unexpected results,
+        # but we've seen the join cluster operation (a POST) fail most often.
+        max_retries = 3
```

Comment:
Looks ok for now, but I guess it could be a kwarg that could default to 3 for GETs and 1 for non-GETs, but then still possible to override from the caller in cases like the join.